lib/index.js

```diff
@@ -0,0 +1,126 @@
+const fs = require('fs');
+const {
+  fetchLabels,
+  createLabels,
+  deleteLabels,
+  updateLabels,
+} = require('./api');
+
+async function labelControl() {
+  const labelData = await fetchLabels();
+  const formattedPublishedLabels = await labelFormatter(labelData);
+  const data = fs.readFileSync('./label-config.json', { encoding: 'utf8' });
+
+  const localConfig = JSON.parse(data.toString());
+  const illegalLabels = checkForSameLabelName(
+    localConfig,
+    formattedPublishedLabels
+  );
+  if (illegalLabels) {
+    throw new Error(
+      'The config has labels with the same name. The Label names have to be unique.'
+    );
+  }
+  const newLabels = checkForNewOrOldLabels(
+    localConfig,
+    formattedPublishedLabels
+  );
+  const labelsToDelete = checkForNewOrOldLabels(
+    formattedPublishedLabels,
+    localConfig
+  );
+  const labelsToUpdate = checkForModifications(
+    localConfig,
+    formattedPublishedLabels
+  );
+  if (newLabels.length !== 0) {
+    await createLabels(newLabels);
+  }
+  if (labelsToDelete.length !== 0) {
+    await deleteLabels(labelsToDelete);
+  }
+  if (labelsToUpdate.length !== 0) {
+    await updateLabels(labelsToUpdate);
+  }
+  await writeLabelsConfig();
+  const changes =
```

Comment:
Since at the end you're treating this variable as a boolean, we might as well assign a boolean to this variable.

```js
const hadChanges =
  newLabels.length > 0 || labelsToDelete.length > 0 || labelsToUpdate > 0;
```

Reply:
💯 
d2e48dc